### ISSUE TITLE
Inconsistent Behavior of XTCReporter Compared to Other Reporters When Target File Already Exists

### ISSUE THREAD

**Issue by z-gong** [openmm/openmm#4651]
When using PDBReporter or DCDReporter, the target file is silently overwritten if it already exists upon initialization. However, with XTCReporter, the behavior is different. The existence of the target .xtc file is only checked when the first set of positions is written, not during initialization. If a file already exists at that point, an exception is raised, and the simulation is terminated.

This behavior can be problematic when a large reporting interval (e.g., 1 million steps) is used. The simulation run for a significant period only to crash once it tries to write to an existing .xtc file.

Proposed Fixes:
- Overwrite the .xtc file if it already exists, making the behavior consistent with PDBReporter and DCDReporter.
- Check for the existence of the .xtc file during the initialization of XTCReporter. If the file exists, raise an exception immediately.

Do you think one of these fixes is reasonable? If so, I’d be happy to draft a pull request.

**Comment by peastman** [comment]
Agreed, we should make the behavior consistent with the other reporters.  A PR would be great!

**Comment by peastman** [comment]
Fixed by #4657.
